lib/pbench/test/functional/server/test_put.py

```diff
@@ -421,6 +432,48 @@ def test_list_filter_and_or(self, server_client: PbenchServerClient, login_user)
                 "2018" in date or "2019" in date
             ), f"Dataset {dataset.name} date is {date}"
 
+    @pytest.mark.dependency(name="list_type", depends=["upload"], scope="session")
+    def test_list_filter_typed(self, server_client: PbenchServerClient, login_user):
+        """Test filtering using typed matches.
+
+        We set an early "server.deletion" time on alternating datasets as we
+        uploaded them. Now prove that we can use a typed "date" filter to find
+        them. We'll further validate that none of the other datasets we
+        uploaded from TARBALL_DIR should have been returned.
+        """
+        test_sets = {}
+        for t in TARBALL_DIR.glob("*.tar.xz"):
+            r = Dataset.md5(t)
+            m = server_client.get_metadata(
+                r,
+                ["dataset.name", "server.deletion"],
+            )
+            test_sets[r] = m
+        soonish = datetime.now(timezone.utc) + timedelta(days=20)
```

Comment:
`20` seems like a magic number.  How about `SHORT_EXPIRATION_DAYS * 2` or `SHORT_EXPIRATION_DAYS + 2`?

Reply:
What do you have against magic? 🪄